pandas/core/arrays/datetimelike.py

```diff
@@ -557,18 +557,8 @@ def astype(self, dtype, copy=True):
             return np.asarray(self, dtype=dtype)
 
     def view(self, dtype=None):
-        """
-        New view on this array with the same data.
-
-        Parameters
-        ----------
-        dtype : numpy dtype, optional
-
-        Returns
-        -------
-        ndarray
-            With the specified `dtype`.
-        """
+        if dtype is None or dtype is self.dtype:
+            return type(self)(self._data, dtype=self.dtype)
         return self._data.view(dtype=dtype)
```

Comment:
This is not returning an EA?

Reply:
correct, the current implementation is only used to return an ndarray.

Reply:
But so that is "violating" the spec? (it should return a new EA (not self), but not an ndarray)

Reply:
yes, but its already in place and used extensively.  I guess we could alter the spec to allow returning ndarray

Reply:
> I guess we could alter the spec to allow returning ndarray

Sorry to further bother on this PR, but I would not alter the spec. For the interface, it should just be a new EA of the same type, no? 

Shouldn't we (ideally, at some point) change our own implementation to return an EA as well for consistency?